src/View/ViewableData.php

```diff
@@ -558,6 +558,17 @@ public function obj($fieldName, $arguments = [], $cache = false, $cacheName = nu
             $value = $this->$fieldName;
         }
 
+        // Wrap arrays
+        if (is_array($value)) {
+            if (array_is_list($value)) {
+                // Wrap in ArrayIterator to respect method signature
+                $value = new ArrayIterator($value);
+            } else {
+                // Wrap in ArrayData so values can be accessed by key in templates
+                $value = ArrayData::create($value);
+            }
+        }
```

Comment:
I've taken a look at the usage of `->obj()` in sink and while a lot of them assume various things, those assumptions either are unaffected or would have already been wrong under certain circumstances anyway.

I'm confident this isn't going to cause any breakage of functionality for anyway, so if I do mention it in the changelog it'll be in passing.